nanoFramework.Tools.DebugLibrary.Shared/Extensions/FlashSectorDataExtensions.cs

```diff
@@ -28,12 +28,26 @@ public static DeploymentSector ToDeploymentSector(this FlashSectorData value)
 
                 // check alignment requirements
                 if ((value.Flags
-                     & BlockRegionAttributes_MASK
-                     & BlockRegionAttribute_ProgramWidthIs64bits) == BlockRegionAttribute_ProgramWidthIs64bits)
+                    & BlockRegionAttributes_MASK
```

Comment:
I like better your previous approach with the switch statement! Cleaner, more maintainable and more efficient.
Here the code keeps doing the same operation ANDing the flags with the mask and it's using consecutive IFs, instead of IF ELSE.

Can we have the switch back please?

And regarding the default, just throw an exception about the aligment value not being supported. That's all. This is to cover future enhancements in the flags and values that need to be caught here. A forward looking safety measure, that's all. 😅

Reply:
Can't throw an exception on default because no bit set is valid alignment of 0.
Am I missing something?